packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -1122,4 +1125,351 @@ describe('KeyringController', () => {
+        SignTypedDataVersion.V4,
+      );
+      const recovered = recoverTypedSignature({
+        data: msgParams,
+        signature,
+        version: SignTypedDataVersion.V4,
+      });
+      expect(account).toBe(recovered);
+    });
+
+    it('should forget Ledger Keyring', async () => {
+      const accounts = await ledgerKeyring.getAccounts();
+      expect(accounts).toHaveLength(1);
+
+      await keyringController.forgetLedger();
+      expect(keyringController.state.keyrings[1].accounts).toHaveLength(0);
+    });
+
+    it('should return Ledger Keyring for corresponding account', async () => {
+      const ledgerAccount = '0xe908e4378431418759b4f87b4bf7966e8aaa5cf2';
+
+      const keyring = await keyringController.getAccountKeyringType(
+        ledgerAccount,
+      );
+
+      expect(keyring).toBe(KeyringTypes.ledger);
+    });
+
+    it('should connect to ledger hardware', async () => {
+      const setTransportStub = sinon.stub(ledgerKeyring, 'setTransport');
+      setTransportStub.resolves();
+
+      const getAppAndVersionStub = sinon.stub(
+        ledgerKeyring,
+        'getAppAndVersion',
+      );
+
+      const mockedAppVersionValue = {
+        appName: 'mockApp',
+        version: 'mockVersion',
+      };
+      getAppAndVersionStub.resolves(mockedAppVersionValue);
+
+      const mockTransport = {} as Transport;
+      const mockDeviceId = 'mockDeviceId';
+
+      const appName = await keyringController.connectLedgerHardware(
+        mockTransport,
+        mockDeviceId,
+      );
+
+      expect(appName).toBe(mockedAppVersionValue.appName);
+    });
+
+    it('should open ethereum app on ledger', async () => {
+      const openEthereumAppOnLedgerStub = sinon.stub(
+        ledgerKeyring,
+        'openEthApp',
+      );
+      openEthereumAppOnLedgerStub.resolves();
+      await keyringController.openEthereumAppOnLedger();
+      expect(openEthereumAppOnLedgerStub.callCount).toBe(1);
+    });
+
+    it('should close running ledger app on ledger', async () => {
+      const getAppAndVersionStub = sinon.stub(
+        ledgerKeyring,
+        'getAppAndVersion',
+      );
+
+      const mockedAppVersionValue = {
+        appName: 'mockApp',
+        version: 'mockVersion',
+      };
+
+      const mockedAppVersionValueBolos = {
+        appName: 'BOLOS',
+        version: 'mockVersion',
+      };
+
+      getAppAndVersionStub.onCall(0).resolves(mockedAppVersionValue);
+      getAppAndVersionStub.onCall(1).resolves(mockedAppVersionValueBolos);
+
+      const quitAppSpy = sinon.stub(ledgerKeyring, 'quitApp');
+      quitAppSpy.resolves();
+
+      await keyringController.closeRunningAppOnLedger();
+      expect(quitAppSpy.callCount).toBe(1);
+
+      // Testing the path when app name is not BOLOS
+      await keyringController.closeRunningAppOnLedger();
+      expect(quitAppSpy.callCount).toBe(1);
+    });
+
+    it('should unlock default account on ledger', async () => {
```

Comment:
It looks like this test checks the return value of this method, but it doesn't test how this method affects state. Can a new test be added to do this? Make sure to check what happens when `!oldAccounts.includes(address)`.

Reply:
I've added another test to check the state change before and after adding the ledger account.